installer.sh

```diff
@@ -0,0 +1,413 @@
+            exit_with_error
+            ;;
+    esac
+
+    output "  [*] Your kernel (${kernel}) is supported" "success"
+}
+
+function check_architecture() {
+    # Detect architecture
+    machine=$(uname -m 2>$pipe || /usr/bin/uname -m)
+    case ${machine} in
+        aarch64*|armv8*|arm64*)
+            machine="arm64"
+            ;;
+        i[36]86|x86)
+            machine="386"
+            ;;
+        x86_64|amd64)
+            machine="amd64"
+            ;;
+        *)
+            output "  [ ] Your architecture (${machine}) is currently not supported" "error"
+            exit_with_error
+            ;;
+    esac
+
+    output "  [*] Your architecture (${machine}) is supported" "success"
+}
+
+function check_install_method() {
+    if [ "linux" == $kernel ]; then
+        if [ -z $PLATFORMSH_CLI_INSTALL_METHOD ]; then
+            PLATFORMSH_CLI_INSTALL_METHOD="raw"
+        fi
+    elif [ "darwin" == $kernel ]; then
+        if [ -z $PLATFORMSH_CLI_INSTALL_METHOD ]; then
+            if command -v brew > /dev/null 2>&1
+            then
+                PLATFORMSH_CLI_INSTALL_METHOD="homebrew"
+            else
+                PLATFORMSH_CLI_INSTALL_METHOD="raw"
+            fi
+        fi
+        if [ $PLATFORMSH_CLI_INSTALL_METHOD = "raw" ]; then
+            machine="all"
+        fi
+    elif [ "freebsd" == $kernel ]; then
+        PLATFORMSH_CLI_INSTALL_METHOD="raw"
+    fi
+
+    output "  [*] Using ${PLATFORMSH_CLI_INSTALL_METHOD} install method" "success"
+}
+
+function init_sudo() {
+    if [ ! -z "${has_sudo}" ]; then
+        return
+    fi
+
+    has_sudo=false
+    # Are we running the installer as root?
+    if [ "$(echo "$UID")" = "0" ]; then
+        has_sudo=true
+        cmd_sudo=''
+
+        return
+    fi
+
+    if command -v sudo > /dev/null 2>&1; then
+        has_sudo=true
+        cmd_sudo='sudo -E'
+    fi
+}
+
+function call_root() {
+    init_sudo
+
+    if ! ${has_sudo}; then
+        output "  sudo is required to perform this operation." "error"
+        exit_with_error
+    fi
+
+    if $cmd_sudo sh -c "$1" 2>&1 | indent; then
+        return 0
+    fi
+
+    return 1
+}
+
+function call_try_user() {
+    if ! call_user "$1"; then
+        output "  command failed, re-trying with sudo" "warning"
+        if ! call_root "$1"; then
+            output "  ${2:-command failed}" "error"
+            exit_with_error
+        fi
+    fi
+}
+
+function call_user() {
+    sh -c "$1" 2>&1 | indent
+}
+
+function check_shasum() {
+    if command -v sha1sum > /dev/null 2>&1; then
+        cmd_shasum="sha1sum"
+        output "  [*] sha1sum is installed" "success"
+    elif command -v shasum > /dev/null 2>&1; then
+        cmd_shasum="shasum -a 1"
+        output "  [*] shasum is installed" "success"
+    else
+        output "  [ ] No sha1sum or shasum available to verify binary" "error"
+        exit_with_error
+    fi
+}
+
+function check_directories() {
```

Comment:
On psh app containers we should use `~/.global/bin`

Reply:
Done, in case of `raw` method, the following directories are considered:

* `$PLATFORMSH_CLI_INSTALL_DIR` - if defined
* `$HOME/.global/bin` - if in `PATH`
* `$HOME/.local/bin` - if in `PATH`
* `/usr/bin` or `/usr/local/bin` for macOS

https://github.com/platformsh/cli/blob/68ece4d3b440e9e26a6da00b1de5d3b8b72efd2a/installer.sh#L311-L317